libres/tests/analysis/test_assert_update_viable.cpp

```diff
@@ -0,0 +1,143 @@
+                          const state_map_type *source_state_map,
+                          const int ensemble_size,
+                          const local_updatestep_type *updatestep);
+}
+
+TEST_CASE("asserting_update_viable", "[analysis][prototype]") {
+
+    GIVEN("Allocated memory and default value parametrs") {
+        // alloc memory for parameters
+        analysis_config_type *analysis_config = analysis_config_alloc_default();
+        state_map_type *source_state_map = state_map_alloc();
+        local_updatestep_type *updatestep =
+            local_updatestep_alloc("name_not_important");
+        int ensemble_size = 1;
+        analysis_module_type *analysis_module =
+            analysis_module_alloc("not-important");
+        local_ministep_type *ministep =
+            local_ministep_alloc("not important", analysis_module);
+        local_updatestep_add_ministep(updatestep, ministep);
+
+        THEN("Assertion fail no active realization") {
+            REQUIRE(analysis_config_get_min_realisations(analysis_config) == 0);
+            REQUIRE(state_map_get_size(source_state_map) == 0);
+            REQUIRE(local_updatestep_get_num_ministep(updatestep) == 1);
+            REQUIRE(!analysis::assert_update_viable(
+                analysis_config, source_state_map, ensemble_size, updatestep));
+        }
+
+        WHEN("State is set to STATE_HAVE_DATA") {
+            state_map_iset(source_state_map, 0, STATE_INITIALIZED);
+            state_map_iset(source_state_map, 0, STATE_HAS_DATA);
+            REQUIRE(state_map_get_size(source_state_map) == 1);
+            REQUIRE(state_map_count_matching(source_state_map,
+                                             STATE_HAS_DATA) == 1);
+            THEN("Assertion pass") {
+                REQUIRE(analysis::assert_update_viable(
+                    analysis_config, source_state_map, ensemble_size,
+                    updatestep));
+            }
+        }
+
+        WHEN("Any state but STATE_HAS_DATA") {
+            realisation_state_enum states[3] = {
+                STATE_INITIALIZED, STATE_LOAD_FAILURE, STATE_PARENT_FAILURE};
+            for (auto state : states) {
+                state_map_iset(source_state_map, 0, STATE_INITIALIZED);
+                state_map_iset(source_state_map, 0, state);
+                REQUIRE(state_map_get_size(source_state_map) == 1);
+                REQUIRE(state_map_count_matching(source_state_map, state) == 1);
+                THEN("Assertion fail no active realization") {
+                    REQUIRE(!analysis::assert_update_viable(
+                        analysis_config, source_state_map, ensemble_size,
+                        updatestep));
+                }
+            }
+        }
+
+        WHEN("one STATE_LOAD_FAILURE, but with others on STATE_HAS_DATA") {
+            for (int i = 0; i < 2; i++) {
+                state_map_iset(source_state_map, i, STATE_INITIALIZED);
```

Comment:
Ok, I'm starting to suspect that the state map first has to be set using `STATE_INITIALIZED` before doing anything else to it 😄 
Would you mind explaining how this works?